nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,45 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async(request, response) => {
+    try {
+        console.log(meals);
+        console.log("in /api/meals");
+        const query = request.query;
+        if (query.maxPrice) {
+            response.send(meals.filter(meal => meal.price <= query.maxPrice));
+        } else if (query.title) {
+            response.send(meals.filter(meal => meal.title.toLowerCase().includes(
+                query.title)));
+        } else if (query.createdAfter) {
+            response.send(meals.filter(meal => new Date(meal.createdAt) > new Date(
+                query.createdAfter)));
+        } else if (query.limit) {
+            response.send(meals.slice(0, query.limit));
+        } else {
+            response.send(meals);
+        }
+    } catch (error) {
+        response.status(400);
+        response.send('Test failed', error);
+    }
+});
+
+router.get("/:id", async(request, response) => {
+    if (!Number.isInteger(parseFloat(request.params.id)) || Number.isNaN(
+            parseFloat(request.params.id))) {
+        response.status(400).send(`Id is not a number`);
+    } else if (parseInt(request.params.id) > meals.length) {
+        response.send(`Id does not exist`);
+    } else {
+        let mealWithId = meals.find(meal => meal.id === parseInt(request.params
+            .id));
+        response.send(mealWithId);
+    }
```

Comment:
This function always returns 400, because this is after the if conditions. You need to place it somewhere else in order to return an error.